Sources/KituraNet/HTTP/HTTPServer.swift

```diff
@@ -158,19 +158,19 @@ public class HTTPServer: Server {
         }
     }
 
-    /// NIOOpenSSL.TLSConfiguration used with the ServerBootstrap
+    /// NIOSSL.TLSConfiguration used with the ServerBootstrap
     private var tlsConfig: TLSConfiguration?
 
     /// The SSLContext built using the TLSConfiguration
-    private var sslContext: NIOOpenSSL.SSLContext?
+    private var sslContext: NIOSSLContext?
 
     /// URI for which the latest WebSocket upgrade was requested by a client
     var latestWebSocketURI: String?
 
     /// Determines if the request should be upgraded and adds additional upgrade headers to the request
-    private func shouldUpgradeToWebSocket(webSocketHandlerFactory: ProtocolHandlerFactory, head: HTTPRequestHead) -> HTTPHeaders? {
+    private func shouldUpgradeToWebSocket(channel: Channel, webSocketHandlerFactory: ProtocolHandlerFactory, head: HTTPRequestHead) -> EventLoopFuture<HTTPHeaders?> {
         self.latestWebSocketURI = head.uri
-        guard webSocketHandlerFactory.isServiceRegistered(at: head.uri) else { return nil }
+        guard webSocketHandlerFactory.isServiceRegistered(at: head.uri) else { return channel.eventLoop.makeSucceededFuture(nil) }
```

Comment:
Why is this future succeeded? If this is an error case shouldn't it be failed?

Reply:
@ianpartridge Yes, it does look wrong. But a failed future needs an Error value to be passed. Here, we're just refusing an upgrade to WebSocket because a service wasn't registered at the requested path. There is no error type defined for this. In the NIO 1 API, the return type of this function `shouldUpgrade` was `HTTPHeaders?`, and we returned a nil to indicate an upgrade refusal. Given this background, even if we had an error type I'm not sure how the WebSocket upgrader would respond to a failed future with an error (I'll take a look at this).

Reply:
From a superficial look at the [WebSocketUpgrader](https://github.com/apple/swift-nio/blob/master/Sources/NIOWebSocket/WebSocketUpgrader.swift#L127) I don't think the upgrader is equipped to handle a failed future with an error(I might be wrong). I propose we take this up through an issue later.

Reply:
OK that's fine.